test/pagination.ts

```diff
@@ -308,3 +308,23 @@ test('ignores the `resolveBodyOnly` option', withServer, async (t, server, got)
 
 	t.deepEqual(items, [1, 2]);
 });
+
+test.failing('allowGetBody sends json payload with .paginate()', withBodyParsingServer, async (t, server, got) => {
+	server.get('/', (request, response) => {
+		if (request.body.hello !== 'world') {
+			response.statusCode = 400;
+		}
+
+		response.end(JSON.stringify([1, 2, 3]));
+	});
+
+	const iterator = got.paginate({
+		allowGetBody: true,
+		json: {hello: 'world'},
+		retry: 0
+	});
+
+	const result = await iterator.next();
+
+	t.is(result.value, 1);
```

Comment:
```suggestion
	t.deepEqual(result.value, [1, 2, 3]);
```

Reply:
otherwise looks good

Reply:
Done. The hack for withBodyParsingServer isn't very nice but I had little time, less inspiration for a better approach.